### ISSUE TITLE
ZOOM: Display the status of the motion safety system on ZOOM

### ISSUE THREAD

**Issue by KathrynBaker** [ISISComputingGroup/IBEX#2934]
As an IBEX user on ZOOM, I want to be able to see the status of the motion safety so that I know why the motors aren't moving.

**Acceptance Criteria**
1. There are PVs available detailing the required information
1. This status is visible in the GUI

**Notes**
1. The hardware is now confirmed as the MOXA E1200, #3269 deals with creating the IOC for interacting with that hardware
1. After discussion, it was agreed that this should be undertaken in a similar architectural style to motion setpoints
1. Values are as follows:
  - E-stops 1-8
  - Enabling switch
  - MC1 (the rack for the SPS, slits and PGC) (i.e. whether the rack is powered ON/OFF)
  - MC2 (the rack for the motion internal to the tank) (i.e. whether the rack is powered ON/OFF)
  - MC3 (the tank and SPS carriage - Beckhoff system) (i.e. whether the rack is powered ON/OFF)

requires #3269

**Comment by KathrynBaker** [comment]
Updating to correct for the change in hardware methodology

**Comment by KathrynBaker** [comment]
#3269 is imminent (hardware being installed).
We should receive the status for the following:

- E-stops 1-8
- Enabling switch
- MC1 (the rack for the SPS, slits and PGC)
- MC2 (the rack for the motion internal to the tank)
- MC3 (the tank and SPS carriage - Beckhoff system)

**Comment by KathrynBaker** [comment]
Updated description to include the content as in the previous comment, and the architectural decision made in the backlog preparation meeting

**Comment by KathrynBaker** [comment]
Alistair and I now have the IP address for the moxa DIO to be used in production, the inputs are as follows:

EMERGENCY STOPS STATUSES
Emergency stop 1 pressed                                           =>           DI0 ON  
Emergency stop 1 released (mechanically reset) =>          DI0 OFF
Emergency stop 2 pressed           =>           DI1 ON
Emergency stop 3 pressed           =>           DI2 ON
Emergency stop 4 pressed           =>           DI3 ON
Emergency stop 5 pressed           =>           DI4 ON
Emergency stop 6 pressed           =>           DI5 ON
Emergency stop 7 pressed           =>           DI6 ON
Emergency stop 8 pressed           =>           DI7 ON

ENABLING SWITCH STATUS
Enabling switch pressed                =>           DI8 ON
Enabling switch released               =>           DI8 OFF

OUTPUT UNITS
Motion Control rack 3 energised               =>           DI9 ON (NOT INSTALLED YET)
Motion Control rack 3 de-energised        =>           DI9 OFF
Motion Control rack 2 energised               =>           DI10 ON
Motion Control rack 1 energised               =>           DI11 ON

PPS PERMITS
Blockhouse searched and locked               =>           DI12 ON
Tank searched and locked = DI13 ON

STATUS OF ALL THE E-STOPS
All emergency stops healthy                       =>           DI14 ON

SPARE
Spare input (NOT IN USE)                             =>           DI15

These macro settings etc. need to be added to ZOOM

**Comment by Alistair-McGann-Tessella** [comment]
When setting the label macros for the DIs I found that the current OPI was truncating the labels. I modified the OPI so that it can display the maximum length of label string. See [commit](https://github.com/ISISComputingGroup/ibex_gui/commit/74be5962a47d6c8feedbc61d0fe0b0073105f2b0).

The regex in the IOC's config.xml was also not inappropriate for the channel strings. This was changed to allow lowercases, space, / and -. See [commit](https://github.com/ISISComputingGroup/EPICS-ioc/commit/28d290050fe6aee3484ec54f9363f25eeaa8a9b6).

These changes were applied to zoom as a hotfix for this ticket. `build_ioc_startups.py` was run to apply the change to the regex.
